File: store-ui/src/app/services/models/product/product-response.ts
import {VariantResponse} from "../variant/variant-response";
import {ProductOverviewResponse} from "./product-overview-response";
import {AttributeResponse} from "../attribute/attribute-response";

export interface ProductResponse {
    id?: number;
    name?: string;
    description?: string;
    price?: number;
    basePrice?: number;
    isPriceEqual?: boolean;
    isVisible?: boolean;
    variants?: VariantResponse[];
    categoryIds?: number[];
    relatedProducts?: ProductOverviewResponse[];
    attributes?: AttributeResponse[];
    imagePaths?: Array<string>;
}

File: services/product/src/main/java/com/ecommerce/product/ProductResponse.java
package com.ecommerce.product;

import com.ecommerce.attribute.AttributeResponse;
import com.ecommerce.variant.VariantResponse;
import lombok.Builder;

import java.math.BigDecimal;
import java.util.List;
import java.util.Set;

@Builder
public record ProductResponse (
        Long id,
        String name,
        String description,
        BigDecimal price,
        BigDecimal basePrice,
        boolean isPriceEqual,
        boolean isVisible,
        Set<VariantResponse> variants,
        Set<ProductOverviewResponse> relatedProducts,
        Set<Long> categoryIds,
        Set<AttributeResponse> attributes,
        List<String> imagePaths
) {}
